tests/src/test/java/alluxio/client/rest/TestCase.java

```diff
@@ -120,15 +109,21 @@ public String getMethod() {
    */
   public URL createURL() throws Exception {
     StringBuilder sb = new StringBuilder();
+    boolean addAmpersands = false;
     for (Map.Entry<String, String> parameter : mParameters.entrySet()) {
-      if (parameter.getValue() == null || parameter.getValue().isEmpty()) {
-        sb.append(parameter.getKey());
-      } else {
-        sb.append(parameter.getKey() + "=" + parameter.getValue() + "&");
+      if (!addAmpersands) { addAmpersands = true; }
+      else { sb.append("&"); }
```

Comment:
I don't like using flag variables for this, but I'm not sure how do achieve this otherwise when using `Map.Entry<>` for-each loops. I assume we'd prefer the boolean flag over converting the entry set into an array and using indices.

Reply:
Is this just adding "&" at the beginning if the parameter is not empty? What about before the for loop:
```
  if (!parameter.isEmpty()) {
    sb.append("&");
  }
```
It's even better if you construct the `url` first, then check for `parameter`, and return before creating the string builder.

Reply:
Oh i think i get the logic backward, lol. You can probably just use guava `Joiner`.

Reply:
Thanks for pointing me to `Joiner` 👍